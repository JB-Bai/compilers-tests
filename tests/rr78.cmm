int P24Dzxkj(){
	float SgcXIe, aFdDTg55_ = 0x3B4a * 0x7, ZZ9bzht = 9878 = 62947, VKSt = 7836.4843 && qxvfjwg(cok_F(! ! 05177 - 0120 / okbw6h < 034747 = 302.518E4 >= 0541.880e-2 && KQgf4Zs.UgBf90oi8 * adCELWtsk || S2oKPu1 + ggSMxr3H6f(66.3164E1 + YxiQwB - 0x8E3 + 12888[208.03e+8 / leU_Arno] != 0x8BE6), 10.70) + 14.10e2 > ! 78.9707 /(00422.xB5GnfC), 0x7fcaA[SvEdhIM()], g8YqTbD);
	int mppNP1, wflNV0Ne = 384.438, Ih1CsXB26[7019], ia5auxs[49335][076767] =(! 0x305[4412 + ! 505[SvDUjm.IXVB83]])[0x398][HptyXK5t][fsOMgS / CSg1olRoCWjx = Mzic(522.5E6 / 995.043, pbc7dIYbS(97, QQHZ1H || 01462[273.52e+7])) && UlggBv4 || 05446];
	return 8728.0;
	- LSqw_B9dvANj(_JjpOkeJ4);
	return - YZ9O2eAf7;
	37.977 - 00640 || NR6A6LqEez - 0x7A && 2.0820E-7 == sTO79aO();
	while(522) bhXEgu + Ph7O9O() * He3kg7;
	Ys4JBfy;
	- R_haq15pVp();
}
float fVivpDDu_, jmE3OLCt4p[7510], CbdGC89g, sgVHv050G;
float ZA6Qud[2687];

