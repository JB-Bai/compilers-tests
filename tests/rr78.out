Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: P24Dzxkj
        LP
        RP
      CompSt (1)
        LC
        DefList (2)
          Def (2)
            Specifier (2)
              TYPE: float
            DecList (2)
              Dec (2)
                VarDec (2)
                  ID: SgcXIe
              COMMA
              DecList (2)
                Dec (2)
                  VarDec (2)
                    ID: aFdDTg55_
                  ASSIGNOP
                  Exp (2)
                    Exp (2)
                      INT: 15178
                    STAR
                    Exp (2)
                      INT: 7
                COMMA
                DecList (2)
                  Dec (2)
                    VarDec (2)
                      ID: ZZ9bzht
                    ASSIGNOP
                    Exp (2)
                      Exp (2)
                        INT: 9878
                      ASSIGNOP
                      Exp (2)
                        INT: 62947
                  COMMA
                  DecList (2)
                    Dec (2)
                      VarDec (2)
                        ID: VKSt
                      ASSIGNOP
                      Exp (2)
                        Exp (2)
                          FLOAT: 7836.484375
                        AND
                        Exp (2)
                          ID: qxvfjwg
                          LP
                          Args (2)
                            Exp (2)
                              Exp (2)
                                Exp (2)
                                  ID: cok_F
                                  LP
                                  Args (2)
                                    Exp (2)
                                      Exp (2)
                                        Exp (2)
                                          Exp (2)
                                            NOT
                                            Exp (2)
                                              NOT
                                              Exp (2)
                                                INT: 2687
                                          MINUS
                                          Exp (2)
                                            Exp (2)
                                              INT: 80
                                            DIV
                                            Exp (2)
                                              ID: okbw6h
                                        RELOP
                                        Exp (2)
                                          INT: 14823
                                      ASSIGNOP
                                      Exp (2)
                                        Exp (2)
                                          Exp (2)
                                            Exp (2)
                                              FLOAT: 3025180.000000
                                            RELOP
                                            Exp (2)
                                              FLOAT: 5.418800
                                          AND
                                          Exp (2)
                                            Exp (2)
                                              Exp (2)
                                                ID: KQgf4Zs
                                              DOT
                                              ID: UgBf90oi8
                                            STAR
                                            Exp (2)
                                              ID: adCELWtsk
                                        OR
                                        Exp (2)
                                          Exp (2)
                                            ID: S2oKPu1
                                          PLUS
                                          Exp (2)
                                            ID: ggSMxr3H6f
                                            LP
                                            Args (2)
                                              Exp (2)
                                                Exp (2)
                                                  Exp (2)
                                                    Exp (2)
                                                      Exp (2)
                                                        FLOAT: 663.164001
                                                      PLUS
                                                      Exp (2)
                                                        ID: YxiQwB
                                                    MINUS
                                                    Exp (2)
                                                      INT: 2275
                                                  PLUS
                                                  Exp (2)
                                                    Exp (2)
                                                      INT: 12888
                                                    LB
                                                    Exp (2)
                                                      Exp (2)
                                                        FLOAT: 20803000320.000000
                                                      DIV
                                                      Exp (2)
                                                        ID: leU_Arno
                                                    RB
                                                RELOP
                                                Exp (2)
                                                  INT: 35814
                                            RP
                                    COMMA
                                    Args (2)
                                      Exp (2)
                                        FLOAT: 10.700000
                                  RP
                                PLUS
                                Exp (2)
                                  FLOAT: 1410.000000
                              RELOP
                              Exp (2)
                                Exp (2)
                                  NOT
                                  Exp (2)
                                    FLOAT: 78.970703
                                DIV
                                Exp (2)
                                  LP
                                  Exp (2)
                                    Exp (2)
                                      INT: 274
                                    DOT
                                    ID: xB5GnfC
                                  RP
                            COMMA
                            Args (2)
                              Exp (2)
                                Exp (2)
                                  INT: 523434
                                LB
                                Exp (2)
                                  ID: SvEdhIM
                                  LP
                                  RP
                                RB
                              COMMA
                              Args (2)
                                Exp (2)
                                  ID: g8YqTbD
                          RP
            SEMI
          DefList (3)
            Def (3)
              Specifier (3)
                TYPE: int
              DecList (3)
                Dec (3)
                  VarDec (3)
                    ID: mppNP1
                COMMA
                DecList (3)
                  Dec (3)
                    VarDec (3)
                      ID: wflNV0Ne
                    ASSIGNOP
                    Exp (3)
                      FLOAT: 384.437988
                  COMMA
                  DecList (3)
                    Dec (3)
                      VarDec (3)
                        VarDec (3)
                          ID: Ih1CsXB26
                        LB
                        INT: 7019
                        RB
                    COMMA
                    DecList (3)
                      Dec (3)
                        VarDec (3)
                          VarDec (3)
                            VarDec (3)
                              ID: ia5auxs
                            LB
                            INT: 49335
                            RB
                          LB
                          INT: 32247
                          RB
                        ASSIGNOP
                        Exp (3)
                          Exp (3)
                            Exp (3)
                              Exp (3)
                                LP
                                Exp (3)
                                  NOT
                                  Exp (3)
                                    Exp (3)
                                      INT: 773
                                    LB
                                    Exp (3)
                                      Exp (3)
                                        INT: 4412
                                      PLUS
                                      Exp (3)
                                        NOT
                                        Exp (3)
                                          Exp (3)
                                            INT: 505
                                          LB
                                          Exp (3)
                                            Exp (3)
                                              ID: SvDUjm
                                            DOT
                                            ID: IXVB83
                                          RB
                                    RB
                                RP
                              LB
                              Exp (3)
                                INT: 920
                              RB
                            LB
                            Exp (3)
                              ID: HptyXK5t
                            RB
                          LB
                          Exp (3)
                            Exp (3)
                              Exp (3)
                                ID: fsOMgS
                              DIV
                              Exp (3)
                                ID: CSg1olRoCWjx
                            ASSIGNOP
                            Exp (3)
                              Exp (3)
                                Exp (3)
                                  ID: Mzic
                                  LP
                                  Args (3)
                                    Exp (3)
                                      Exp (3)
                                        FLOAT: 522500000.000000
                                      DIV
                                      Exp (3)
                                        FLOAT: 995.043030
                                    COMMA
                                    Args (3)
                                      Exp (3)
                                        ID: pbc7dIYbS
                                        LP
                                        Args (3)
                                          Exp (3)
                                            INT: 97
                                          COMMA
                                          Args (3)
                                            Exp (3)
                                              Exp (3)
                                                ID: QQHZ1H
                                              OR
                                              Exp (3)
                                                Exp (3)
                                                  INT: 818
                                                LB
                                                Exp (3)
                                                  FLOAT: 2735200000.000000
                                                RB
                                        RP
                                  RP
                                AND
                                Exp (3)
                                  ID: UlggBv4
                              OR
                              Exp (3)
                                INT: 2854
                          RB
              SEMI
        StmtList (4)
          Stmt (4)
            RETURN
            Exp (4)
              FLOAT: 8728.000000
            SEMI
          StmtList (5)
            Stmt (5)
              Exp (5)
                MINUS
                Exp (5)
                  ID: LSqw_B9dvANj
                  LP
                  Args (5)
                    Exp (5)
                      ID: _JjpOkeJ4
                  RP
              SEMI
            StmtList (6)
              Stmt (6)
                RETURN
                Exp (6)
                  MINUS
                  Exp (6)
                    ID: YZ9O2eAf7
                SEMI
              StmtList (7)
                Stmt (7)
                  Exp (7)
                    Exp (7)
                      Exp (7)
                        FLOAT: 37.977001
                      MINUS
                      Exp (7)
                        INT: 416
                    OR
                    Exp (7)
                      Exp (7)
                        Exp (7)
                          ID: NR6A6LqEez
                        MINUS
                        Exp (7)
                          INT: 122
                      AND
                      Exp (7)
                        Exp (7)
                          FLOAT: 0.000000
                        RELOP
                        Exp (7)
                          ID: sTO79aO
                          LP
                          RP
                  SEMI
                StmtList (8)
                  Stmt (8)
                    WHILE
                    LP
                    Exp (8)
                      INT: 522
                    RP
                    Stmt (8)
                      Exp (8)
                        Exp (8)
                          ID: bhXEgu
                        PLUS
                        Exp (8)
                          Exp (8)
                            ID: Ph7O9O
                            LP
                            RP
                          STAR
                          Exp (8)
                            ID: He3kg7
                      SEMI
                  StmtList (9)
                    Stmt (9)
                      Exp (9)
                        ID: Ys4JBfy
                      SEMI
                    StmtList (10)
                      Stmt (10)
                        Exp (10)
                          MINUS
                          Exp (10)
                            ID: R_haq15pVp
                            LP
                            RP
                        SEMI
        RC
    ExtDefList (12)
      ExtDef (12)
        Specifier (12)
          TYPE: float
        ExtDecList (12)
          VarDec (12)
            ID: fVivpDDu_
          COMMA
          ExtDecList (12)
            VarDec (12)
              VarDec (12)
                ID: jmE3OLCt4p
              LB
              INT: 7510
              RB
            COMMA
            ExtDecList (12)
              VarDec (12)
                ID: CbdGC89g
              COMMA
              ExtDecList (12)
                VarDec (12)
                  ID: sgVHv050G
        SEMI
      ExtDefList (13)
        ExtDef (13)
          Specifier (13)
            TYPE: float
          ExtDecList (13)
            VarDec (13)
              VarDec (13)
                ID: ZA6Qud
              LB
              INT: 2687
              RB
          SEMI
